extensions/binding/org.eclipse.smarthome.binding.onewire/src/main/java/org/eclipse/smarthome/binding/onewire/internal/handler/EDSSensorThingHandler.java

```diff
@@ -0,0 +1,219 @@
+
+import static org.eclipse.smarthome.binding.onewire.internal.OwBindingConstants.*;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.binding.onewire.internal.OwDynamicStateDescriptionProvider;
+import org.eclipse.smarthome.binding.onewire.internal.OwException;
+import org.eclipse.smarthome.binding.onewire.internal.OwPageBuffer;
+import org.eclipse.smarthome.binding.onewire.internal.device.EDS006x;
+import org.eclipse.smarthome.binding.onewire.internal.device.OwSensorType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EDSSensorThingHandler} is responsible for handling EDS multisensors
+ *
+ * @author Jan N. Klug - Initial contribution
+ */
+@NonNullByDefault
+public class EDSSensorThingHandler extends OwBaseThingHandler {
+    public static final Set<ThingTypeUID> SUPPORTED_THING_TYPES = new HashSet<>(Arrays.asList(THING_TYPE_EDS_ENV));
+
+    private final Logger logger = LoggerFactory.getLogger(EDSSensorThingHandler.class);
+
+    private OwSensorType sensorType = OwSensorType.UNKNOWN;
+
+    public EDSSensorThingHandler(Thing thing, OwDynamicStateDescriptionProvider dynamicStateDescriptionProvider) {
+        super(thing, dynamicStateDescriptionProvider);
+    }
+
+    @Override
+    public void initialize() {
+        Map<String, String> properties = editProperties();
+
+        if (!super.configure()) {
+            return;
+        }
+
+        if (getThing().getStatus() == ThingStatus.OFFLINE) {
+            return;
+        }
+
+        // add sensors
+        sensors.add(new EDS006x(sensorIds.get(0), this));
+
+        // check sensors
+        if (!properties.containsKey(PROPERTY_MODELID)) {
+            updateSensorProperties();
+            return;
+        } else {
+            sensorType = OwSensorType.valueOf(properties.get(PROPERTY_MODELID));
+            ((EDS006x) sensors.get(0)).configureChannels(sensorType);
+        }
+
+        scheduler.execute(() -> {
+            configureThingChannels();
+        });
+    }
+
+    private void configureThingChannels() {
+        ThingBuilder thingBuilder = editThing();
+
+        logger.debug("configuring sensors for {}", this.thing.getLabel());
+
+        try {
+            ((EDS006x) sensors.get(0)).configureChannels(sensorType);
```

Comment:
Why not calling `sensors.get(0)` once and store the result in a local variable?
In the different switch cases `sensor.get(0)` is called multiple times (for EDS0068 six times), I don't think the sensor differs between the calls, but the lookup needs to be done every time.